client/db/src/stats.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Database usage statistics
+
+use std::sync::atomic::{AtomicU64, Ordering as AtomicOrdering};
+
+/// Accumulated usage statistics for state queries.
+pub struct StateUsageStats {
+	started: std::time::Instant,
+	reads: AtomicU64,
+	bytes_read: AtomicU64,
+	writes: AtomicU64,
+	bytes_written: AtomicU64,
+	reads_cache: AtomicU64,
+	bytes_read_cache: AtomicU64,
+}
```

Comment:
Maybe add `reads/writes` to `telemetry` and `record_metrics` if they are missing.

Reply:
Yep, I'll open another PR for that when I figure out telemetry, ok?

Reply:
Ah, isn't it just adding the lines here? https://github.com/paritytech/substrate/blob/4ed0ad6fbd8b25b63105dc21710ace3c329de0d1/client/service/src/builder.rs#L968

Reply:
Yeah I added previous metrics there, just unsure if anything in telemetry needs or displays those

Reply:
I see, I know that for `record_metrics` is ok, and can be displayed using grafana.